distributed/client.py

```diff
@@ -130,6 +130,13 @@
 TOPIC_PREFIX_FORWARDED_LOG_RECORD = "forwarded-log-record"
 
 
+class SourceCode(NamedTuple):
+    code: str
+    lineno_frame: int
+    lineno_relative: int
```

Comment:
nit: I'd personally prefer having the absolute number here. This would match the `frame` from `inspect` and slightly facilitate the logic. Also it would make it easier for _users_ to jump to the correct line in the source code. Feel free to ignore.